sdk/iot/azure-iot-modelsrepository/azure/iot/modelsrepository/_constants.py

```diff
@@ -0,0 +1,12 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import platform
+
+VERSION = "0.0.0-preview"
```

Comment:
This will not be a compliant version string. We should go with `1.0.0b1` or `1.0.0a1` which are compatible with pep440 and sdk guidelines. When we iterate on the preview SDK we can increment the pre-release segment. This is also aligned with SDKs for Java and dotnet.

Reply:
Python will automatically convert the string "preview" into "rc" which is PEP440 compliant. So this semantic version would be 1.0.0-preview.1 (i.e. 1.0.0-rc1)

"rc" indicates "release candidate". Do you have a preference between the semantics of "alpha"/"beta"/"release candidate"?

Reply:
Rather than rely on packaging to convert a non-pep440 compliant pre-release segment to a compliant one, it would be better to commit a compliant one in source code.

pep440 has a section on [semantic versioning](https://www.python.org/dev/peps/pep-0440/#semantic-versioning) which states "Semantic versions containing a hyphen (pre-releases - clause 10) or a plus sign (builds - clause 11) are not compatible with this PEP and are not permitted in the public version field."

The azure SDK guidelines also state:

![image](https://user-images.githubusercontent.com/28658112/115600797-e9b76500-a291-11eb-9a6f-69195b04202f.png)

Furthermore we can compare with prior-art with respect to pre-release packages in the track 2 mono repo. The preview azure quantum mgmt SDK release is versioned as [`1.0.0b1`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/quantum/azure-mgmt-quantum/azure/mgmt/quantum/_version.py)

Another example with the preview azure container registry package is versioned as [`1.0.0b1`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/containerregistry/azure-containerregistry/azure/containerregistry/_version.py)

Given the above reasons I would prefer an explicit `1.0.0b1`, `1.0.0a1`, or `1.0.0rc1` with my most preferred being `1.0.0b1`.